org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java

```diff
@@ -86,7 +89,7 @@ public Image getImage(ImageDescriptor descriptor) {
 		}
 		result = descriptor.createImage();
 		if (result != null) {
-			registry.put(descriptor, result);
+			putToRegistry(descriptor, result);
```

Comment:
It's all quite nasty:
`getImage(..)` is also being called for "foreign" image descriptors from other bundles. Since we put the respective images into the local registry, it's probably fine to dispose these, too.

Leaking the helper can be avoided by using a static inner class where we nullify the reference to the registry when the bundle is being disposed.

Reply:
hmm i didnt get this one. can you please elaborate?

Reply:
i will check if the dispose on display will null the listner

Reply:
yes it acutally does. so it should not be a problem.

Reply:
So the remaining theoretical problem is, that the display will keep a reference to the registry if the bundle is being disposed. Disposing a bundle cleanly is out of scope I guess.

Reply:
ahh now i also get the null out point. will think about it.

Reply:
what do you think about https://github.com/eclipse/xtext-eclipse/compare/cd_disposedFix_variantB

Reply:
updated to use a static inner classes and two fold nulling